### PR TITLE
JoeyNMT version2.0.0

### PR DESCRIPTION
**Pull request by may-**
- upgrade to python 3.9, torch 1.11
- `torchtext.legacy` dependencies are completely replaced by `torch.utils.data`
- `joeynmt/tokenizers.py`: handles tokenization internally (also supports bpe-dropout!)
- `joeynmt/datasets.py`: loads data from plaintext, tsv, and huggingface's [datasets](https://github.com/huggingface/datasets)
- `scripts/build_vocab.py`: trains subwords, creates joint vocab
- enhancement in decoding
  - scoring with hypotheses or references
  - repetition penalty, ngram blocker
  - attention plots for transformers
- yapf, isort, flake8 introduced
- bugfixes, minor refactoring

### CODE REVIEW

**Comment by juliakreutzer** [comment]
THANKS A TON!!! :partying_face:

File: setup.py [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 
 with open("requirements.txt", encoding="utf-8") as req_fp:
     install_requires = req_fp.readlines()
 
 setup(
     name='joeynmt',
-    version='1.5.1',
+    version='2.0.0',
     description='Minimalist NMT for educational purposes',
     author='Jasmijn Bastings and Julia Kreutzer',
```
Comment (juliakreutzer) [inline setup.py:11]: and Mayumi Ohta :)
